library/lua/dfhack.lua

```diff
@@ -417,12 +476,23 @@ end
 
 -- String class extentions
 
+---@class string
+local string
```

Comment:
this is why the tests are failing. this is overriding the builtin string definition

Reply:
Figured, the language server complains about the overridden functions otherwise... I'll have to play around with this.